Automattic/wp-calypso
# Add platform and hosting check, add hosting error message, continue anyway path for WPCOM sites

Related to https://github.com/Automattic/wp-calypso/issues/94915

## Proposed Changes

* Added functionality to fetch platform and hosting info for source site.
* Added error message for displaying if site's already on WordPress.com.
* Also, we won't need any further checks to implement to direct users to either the survey page or other platform importing pages. We'll just need to direct them when we have the pages ready.

## Why are these changes being made?


* To let the users know that they are already on WordPress.com

## Testing Instructions



- Proceed through the migration flow. Make sure to give input to a site that's already on WordPress.com
- On the "How do you want to migrate?" step, choose "Do it for me".
- In the credentials form, fill the fields and click on Continue button. Check for the following cases-

#### Site already on WordPress.com
- Make sure you see an error below the source URL field to remind you that the site is already on WordPress.com
- Make sure the Continue button has changed to "Continue anyways". Click that button
- Make sure your request completes as expected and you are taken to the next page

#### WP site, but not on WordPress.com
- Now repeat the whole flow with a site that's not on wordpress.com, but is still a valid Wordpress site. Once you click on the Continue button, you shouldn't see any error (if your creds are valid) and your request should complete as expected.

#### Non-WP site
- Now repeat the whole process with a site that's not a WordPress site at all. Once you click on the Continue button, you shouldn't see similar validation and "Continue anyways" behavior of the button as before

https://github.com/user-attachments/assets/5d32eabb-8d60-4f52-8c9e-1c738508a9a5

## Pre-merge Checklist
